RevenueCatUI/Templates/Components/TemplateComponentsView.swift

```diff
@@ -51,14 +50,34 @@ public struct TemplateComponentsView: View {
         .edgesIgnoringSafeArea(.top)
     }
 
-}
+    static func chooseLocalization(
+        for componentsData: PaywallComponentsData
+    ) -> (locale: Locale, localizedStrings: PaywallComponent.LocalizationDictionary) {
 
-func getLocalization(_ locale: Locale, _ displayString: DisplayString) -> String {
-    if let found = displayString.value[locale.identifier] {
-        return found
-    }
+        guard !componentsData.componentsLocalizations.isEmpty else {
+            Logger.error(Strings.paywall_contains_no_localization_data)
+            return (Locale.current, PaywallComponent.LocalizationDictionary())
+        }
+
+        // STEP 1: Get available paywall locales
+        let paywallLocales = componentsData.componentsLocalizations.keys.map { Locale(identifier: $0) }
 
-    return displayString.value.values.first!
+        let fallbackLocale = Locale(identifier: componentsData.defaultLocale)
+
+        // STEP 2: choose best locale based on device's list of preferred locales.
+        let chosenLocale = Self.preferredLocale(from: paywallLocales) ?? fallbackLocale
+
+        // STEP 3: Get localization for one of preferred locales in order
+        if let localizedStrings = componentsData.componentsLocalizations[chosenLocale.identifier] {
+            return (chosenLocale, localizedStrings)
+        } else if let localizedStrings = componentsData.componentsLocalizations[fallbackLocale.identifier] {
+            Logger.error(Strings.paywall_could_not_find_localization("\(chosenLocale)"))
```

Comment:
Need to log the correct locale (and I made the suggestion `defaultLocale` like my suggestion above ☝️ 

```suggestion
            Logger.error(Strings.paywall_could_not_find_localization("\(defaultLocale)"))
```

Reply:
I think it's correct as written? It will only enter this block if it failed to find `chosenLocale.identifier` in the first if let statement. At this point, it has found the localization for `defaultLocale`.

This would be easier to read if it used successive if statements rather than using elses, so I can make that change.

Reply:
Oh yeaaahhh hahaha... I read it wrong 😅  Nevermind!